test/e2e/remediation.go

```diff
@@ -136,7 +136,7 @@ func remediation(ctx context.Context, inputGetter func() RemediationInput) {
 
 	By("Testing unhealthy and inspection annotations")
 	By("Scaling down KCP to 1 replica")
-	newReplicaCount := 1
+	newReplicaCount := int32(1)
```

Comment:
This is the lone test case that has variable instead of just harcoded number. 

It is questionable why it is int32 param in the first place, why not uint32 or just uint? If we change that, its going to be follow-up tho.